### PR TITLE
Processing algorithms run with missing field

### PR DESCRIPTION
**Pull request by koebi**
Following 7e95813a, this is now also fixed in all other processing
algorithms.

### CODE REVIEW

File: ORStools/proc/directions_lines_proc.py [hunk 1]
```diff
@@ -102,13 +104,18 @@ def processAlgorithm(self, parameters, context, feedback):
...
+        # def __bool__(self):
+        #     return not self.isNull()
+        if source_field_name := parameters[self.IN_FIELD]:
+            sink_fields = directions_core.get_fields(
+                    from_type=source.fields().field(source_field_name).type(),
+                    from_name=source_field_name, line=True)
+        else:
+            sink_fields = directions_core.get_fields(line=True)
+
+        (sink, dest_id) = self.parameterAsSink(parameters, self.OUT, context, sink_fields,
```
Comment (TheGreatRefrigerator) [inline ORStools/proc/directions_lines_proc.py:118]: maybe fill dynamically (no if-else, but only add options if)
```suggestion
        source_field_name = parameters[self.IN_FIELD]

        get_field_options = {
            "line": True
        }
        if source_field_name:
            get_field_options.update(
                from_type=source.fields().field(source_field_name).type(),
                from_name=source_field_name
            )

        sink_fields = directions_core.get_fields(**get_field_options)

        (sink, dest_id) = self.parameterAsSink(parameters, self.OUT, context, sink_fields,
```

File: ORStools/proc/directions_lines_proc.py [hunk 2]
```diff
@@ -167,7 +174,10 @@ def _get_sorted_lines(layer, field_name):
 
         for feat in sorted(layer.getFeatures(), key=lambda f: f.id()):
             line = None
-            field_value = feat[field_name]
+            try:
+                field_value = feat[field_name]
+            except TypeError:
+                field_value = None
```
Comment (TheGreatRefrigerator) [inline ORStools/proc/directions_lines_proc.py:180]: do we need a try here? (or also just if else as above)
```suggestion
            field_value = feat[field_name] if field_name else None
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -48,6 +48,7 @@ RELEASING:
 - author information
 - repository link
 - Correct isochrone computation from layer without fields
+- other errors stemming from layers without fields
```
Comment (TheGreatRefrigerator) [inline CHANGELOG.md:51]: also add to metadata.txt

File: ORStools/proc/directions_points_layer_proc.py [hunk 4]
```diff
@@ -102,13 +104,18 @@ def processAlgorithm(self, parameters, context, feedback):
...
-        source_field_name = parameters[self.IN_FIELD]
+        if source_field_name := parameters[self.IN_FIELD]:
+           sink_fields = directions_core.get_fields(
+                    from_type=source.fields().field(source_field_name).type(),
+                    from_name=source_field_name,
+                    line=True
+                )
+        else:
+            sink_fields = directions_core.get_fields(line=True)
+
```
Comment (TheGreatRefrigerator) [inline ORStools/proc/directions_points_layer_proc.py:115]: as above

File: ORStools/proc/directions_points_layer_proc.py [hunk 5]
```diff
@@ -130,7 +137,10 @@ def processAlgorithm(self, parameters, context, feedback):
                 for point in feat.geometry().asMultiPoint():
                     points.append(x_former.transform(QgsPointXY(point)))
                 input_points.append(points)
-                from_values.append(feat[source_field_name])
+                try:
+                    from_values.append(feat[source_field_name])
+                except KeyError:
+                    from_values.append(None)
```
Comment (TheGreatRefrigerator) [inline ORStools/proc/directions_points_layer_proc.py:143]: as above

File: ORStools/proc/matrix_proc.py [hunk 6]
```diff
@@ -97,19 +101,20 @@ def processAlgorithm(self, parameters, context, feedback):
             self.IN_START,
             context
         )
-        source_field_name = parameters[self.IN_START_FIELD]
+        if source_field_name := parameters[self.IN_START_FIELD]:
```
Comment (TheGreatRefrigerator) [inline ORStools/proc/matrix_proc.py:104]: as above, no walrus then maybe if else short hand assignment

File: ORStools/proc/matrix_proc.py [hunk 7]
```diff
@@ -97,19 +101,20 @@ def processAlgorithm(self, parameters, context, feedback):
...
             self.IN_END,
             context
         )
-        destination_field_name = parameters[self.IN_END_FIELD]
-
-        # Get fields from field name
-        source_field = source.fields().field(source_field_name)
-
-        destination_field = destination.fields().field(destination_field_name)
+        if destination_field_name := parameters[self.IN_END_FIELD]:
```
Comment (TheGreatRefrigerator) [inline ORStools/proc/matrix_proc.py:114]: you know what

**Review by TheGreatRefrigerator — CHANGES_REQUESTED** [review]
Orstools/proc/isochrones_layer_proc should be structured similarly, as it is currenlty breaking because of :
https://github.com/GIScience/orstools-qgis-plugin/blob/master/ORStools/proc/isochrones_layer_proc.py#L125
If we allow empty id we don't need to set by default.

**Comment by TheGreatRefrigerator** [comment]
Please also rebase from master <3

File: ORStools/proc/directions_lines_proc.py [hunk 8]
```diff
@@ -102,13 +104,18 @@ def processAlgorithm(self, parameters, context, feedback):
...
-
-        (sink, dest_id) = self.parameterAsSink(parameters, self.OUT, context,
-                                               directions_core.get_fields(
-                                                   from_type=source.fields().field(source_field_name).type(),
-                                                   from_name=source_field_name,
-                                                   line=True),
+        # qgis overwrites PyQt5.QtCore.QVariant's __bool__ in
+        # https://github.com/qgis/QGIS/blob/master/python/PyQt/PyQt5/QtCore.py:
+        # def __bool__(self):
+        #     return not self.isNull()
```
[the remark this answers is not in the document]
Comment (koebi) [reply ORStools/proc/directions_lines_proc.py:110]: I'll expand on that comment a bit :)

File: ORStools/proc/directions_lines_proc.py [hunk 9]
```diff
@@ -102,13 +104,18 @@ def processAlgorithm(self, parameters, context, feedback):
...
+        if source_field_name := parameters[self.IN_FIELD]:
```
[the remark this answers is not in the document]
Comment (koebi) [reply ORStools/proc/directions_lines_proc.py:111]: agree, I'll change that

File: ORStools/proc/directions_lines_proc.py (hunk 1 above)
Comment (koebi) [reply ORStools/proc/directions_lines_proc.py]: done - a bit differently though, as `line = True` is always set.

File: ORStools/proc/directions_lines_proc.py (hunk 2 above)
Comment (koebi) [reply ORStools/proc/directions_lines_proc.py:185]: no, we don't. fixed that

File: CHANGELOG.md (hunk 3 above)
Comment (koebi) [reply CHANGELOG.md:55]: done

File: ORStools/proc/directions_points_layer_proc.py (hunk 4 above)
Comment (koebi) [reply ORStools/proc/directions_points_layer_proc.py]: done

File: ORStools/proc/directions_points_layer_proc.py (hunk 5 above)
Comment (koebi) [reply ORStools/proc/directions_points_layer_proc.py]: done

File: ORStools/proc/matrix_proc.py (hunk 6 above)
Comment (koebi) [reply ORStools/proc/matrix_proc.py]: done

File: ORStools/proc/matrix_proc.py (hunk 7 above)
Comment (koebi) [reply ORStools/proc/matrix_proc.py]: done

**Comment by koebi** [comment]
> Please also rebase from master <3

done

File: ORStools/proc/directions_points_layer_proc.py [hunk 10]
```diff
@@ -103,12 +105,18 @@ def processAlgorithm(self, parameters, context, feedback):
...
 
         source_field_name = parameters[self.IN_FIELD]
+        if source_field_name:
+           get_fields_options = dict(
+                    from_type=source.fields().field(source_field_name).type(),
+                    from_name=source_field_name
+                )
+        else:
+
+        sink_fields = directions_core.get_fields(**get_fields_options, line=True)
```
Comment (TheGreatRefrigerator) [inline ORStools/proc/directions_points_layer_proc.py:115]: 😅 did you actually test after adjusting?
-> indentation error 😝 
(pycharm would have noticed you....)

There is missing something here even if indented, if: defines `get_fields_options`, else: defines `sink_fields` ?
like this `get_fields_options` is not initialized if no `source_field_name` is given

File: ORStools/proc/directions_points_layers_proc.py [hunk 11]
```diff
@@ -113,18 +117,18 @@ def processAlgorithm(self, parameters, context, feedback):
...
             parameters,
             self.IN_END,
             context
         )
-        destination_field_name = parameters[self.IN_END_FIELD]
 
-        # Get fields from field name
-        source_field = source.fields().field(source_field_name)
-        destination_field = destination.fields().field(destination_field_name)
+        destination_field_name = parameters[self.IN_START_FIELD]:
```
Comment (TheGreatRefrigerator) [inline ORStools/proc/directions_points_layers_proc.py:130]: no :

File: ORStools/proc/directions_lines_proc.py [hunk 12]
```diff
@@ -102,13 +104,23 @@ def processAlgorithm(self, parameters, context, feedback):
...
+        # parameters[self.IN_FIELD] returns a PyQt5.QtCore.QVariant with "NULL" as content
+        # in case of absence of self.IN_FIELD.
+        # qgis overwrites this type's __bool__ in
+        # https://github.com/qgis/QGIS/blob/master/python/PyQt/PyQt5/QtCore.py:
+        # def __bool__(self):
+        #     return not self.isNull()
+        # The check below works because of that.
         source_field_name = parameters[self.IN_FIELD]
+        if source_field_name:
+            get_field_options = dict(
```
Comment (TheGreatRefrigerator) [inline ORStools/proc/directions_lines_proc.py:116]: `get_field_options` is not initialized like this if `source_field_name` is false

File: ORStools/proc/isochrones_layer_proc.py [hunk 13]
```diff
@@ -122,20 +122,11 @@ def processAlgorithm(self, parameters, context, feedback):
 
         # Get ID field properties
         id_field_name = parameters[self.IN_FIELD]
-        id_field_id = source.fields().indexOf(id_field_name)
+        if id_field_name:
+            id_field = source.fields().field(id_field_name)
+            parameter_options = [id_field.type(), id_field_name]
```
Comment (TheGreatRefrigerator) [inline ORStools/proc/isochrones_layer_proc.py:127]: referenced before assignment for `id_field_name` = false

**Review by TheGreatRefrigerator — CHANGES_REQUESTED** [review]
pls test 🧪

File: ORStools/proc/directions_lines_proc.py (hunk 2 above)
Comment (TheGreatRefrigerator) [reply ORStools/proc/directions_lines_proc.py:185]: not yet fixed in that file

File: ORStools/proc/isochrones_layer_proc.py (hunk 13 above)
Comment (koebi) [reply ORStools/proc/isochrones_layer_proc.py:128]: done

File: ORStools/proc/directions_lines_proc.py (hunk 12 above)
Comment (koebi) [reply ORStools/proc/directions_lines_proc.py]: done

File: ORStools/proc/directions_points_layers_proc.py (hunk 11 above)
Comment (koebi) [reply ORStools/proc/directions_points_layers_proc.py]: done

File: ORStools/proc/directions_points_layer_proc.py (hunk 10 above)
Comment (koebi) [reply ORStools/proc/directions_points_layer_proc.py:115]: …I didn't :see_no_evil: 
Should be fixed :)

**Comment by koebi** [comment]
This time, I ran all algorithms :)

File: ORStools/proc/directions_lines_proc.py (hunk 2 above)
Comment (koebi) [reply ORStools/proc/directions_lines_proc.py]: and done :)

**Review by TheGreatRefrigerator — APPROVED** [review]

**Pull request merged**